client/allocrunner/taskrunner/plugin_supervisor_hook.go

```diff
@@ -103,6 +103,16 @@ func newCSIPluginSupervisorHook(config *csiPluginSupervisorHookConfig) *csiPlugi
 	socketMountPoint := filepath.Join(config.clientStateDirPath, "csi",
 		"plugins", config.runner.Alloc().ID)
 
+	// In v1.3.0, Nomad started instructing CSI plugins to stage and publish 
+	// within /csi/local. Plugins deployed after the introduction of
+	// StagePublishDir default to StagePublishDir = /csi/local. However, 
+	// plugins deployed between v1.3.0 and the introduction of 
+	// StagePublishDir have StagePublishDir = "". Default to /csi/local here
+	// to avoid breaking plugins that aren't redeployed.
+	if task.CSIPluginConfig.StagePublishDir == "" {
+		task.CSIPluginConfig.StagePublishDir = filepath.Join("/local", "csi")
+	}
```

Comment:
I ran into an issue while upgrading in my own environment from 1.3.x to a developer build of my changes. I'm not sure if there's a better (more canonical) approach to making sure CSI plugins don't need to be redeployed. This may also be a location where I need to add testing?